src/internet_identity/src/storage.rs

```diff
@@ -140,46 +194,46 @@ impl<T: candid::CandidType + serde::de::DeserializeOwned> Storage<T> {
     }
 
     /// Writes the data of the specified user to stable memory.
-    pub fn write(&self, user_number: UserNumber, data: T) -> Result<(), StorageError> {
+    pub fn write(&mut self, user_number: UserNumber, data: T) -> Result<(), StorageError> {
         let record_number = self.user_number_to_record(user_number)?;
 
-        let stable_offset = HEADER_SIZE + record_number as u64 * self.header.entry_size as u64;
+        let stable_offset =
+            RESERVED_HEADER_BYTES + record_number as u64 * self.header.entry_size as u64;
         let buf = candid::encode_one(data).map_err(StorageError::SerializationError)?;
 
         if buf.len() > self.value_size_limit() {
             return Err(StorageError::EntrySizeLimitExceeded(buf.len()));
         }
 
-        let current_size = stable64_size();
-        let pages =
-            (stable_offset + self.header.entry_size as u64 + WASM_PAGE_SIZE - 1) / WASM_PAGE_SIZE;
-        if pages > current_size {
-            let pages_to_grow = pages - current_size;
-            let result = stable64_grow(pages - current_size);
-            if result.is_err() {
-                trap(&format!(
-                    "failed to grow stable memory by {} pages",
-                    pages_to_grow
-                ))
-            }
-        }
```

Comment:
Guess all the growing business is now hidden away in `Memory`, correct?

Reply:
It's hidden in `Writer`, but yes.